flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/local/DependencyTest.java

```diff
@@ -193,7 +200,20 @@ public Catalog createCatalog(String name, Map<String, String> properties) {
 					ADDITIONAL_TEST_DATABASE,
 					new CatalogDatabaseImpl(new HashMap<>(), null),
 					false);
-			} catch (DatabaseAlreadyExistException e) {
+				hiveCatalog.createTable(
```

Comment:
@bowenli86 please remove the Hive dependency from the SQL Client. For tests would should follow a clear separation of concerns. We just want to test catalog functionality so we can implement a dummy catalog. Combining multiple modules to test end-to-end behavior should be done in an end-to-end test. Otherwise we end up with complex module dependency structures that are actually not necessary for testing.

Reply:
yeah, the reason at that time is we don't have a good way to test and cover catalog discovery for `HiveCatalog`. As soon as we have end-2-end test (WIP https://github.com/apache/flink/pull/9149), we should be able to remove hive test dependencies from sql cli

Reply:
But you don't need a Hive catalog to test catalog functionality. You can simply create a test catalog similar to sources and sinks in this test.

Reply:
we have a plan to remove the dependencies